static/src/javascripts/projects/commercial/modules/dfp/fill-advert-slots.ts

```diff
@@ -66,11 +66,14 @@ const fillAdvertSlots = async (): Promise<void> => {
 				const advert = new Advert(adSlot, additionalSizes);
 				return advert;
 			} catch {
+				log(
```

Comment:
Would this be a duplication of the error handling in `Advert.ts`?

Is an option to catch, report but _not_ throw from `Advert` and keep the filter here so invalid ads do not progress?

Reply:
Yep, good point. I agree that it makes sense to catch and log the error in only one place.

I think the catch should be above `Advert`, since once an error has been thrown, we do not want to continue normal processing of creating the Advert object. I think this makes more sense than returning an invalid `Advert` object.

With regard to Sentry, I was worried that logging this kind of advert failure to Sentry would result in hitting our rate limits, so I removed it. However, I then found this piece of code that logs empty adverts to Sentry at a rate of 1 in every 10,000: https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts#L19 

Do you think I should add back in the log to Sentry at a similar rate to the one above, or a more frequent rate, or not at all?

Reply:
I think we should report the error to Sentry with sampling - perhaps by updating `reportError` to take an optional sampling function parameter?

Sentry will catch any thrown exceptions but as we are in a loop I think we'd want to continue filling adverts so we should catch and report?

Reply:
Is 1 in every 10,000 sensible or maybe higher/lower? It's really hard to tell right now how often this is happening. 

When it throws currently, does it appear it Sentry? I could not find the error when I looked yesterday.

Reply:
Looking at [raven.js](https://github.com/guardian/frontend/blob/fbe9b761be09eed900477b77df67e207db0e14a4/static/src/javascripts/lib/raven.js#L55) it would appear all errors to Sentry are sampled at 0.08% so I think sampling automatically happens when you call `reportError`.

Which means the sampling in `on-slot-render` is an intentional further sampling decrease or a mistake :) 

I'd stick with the existing sampling for now and review Sentry after release.

With the traffic we get we should start to see errors coming through if this is a problem.

Reply:
Ah OK I hadn't realised these error were already sampled at 0.08%. That makes sense. I'll remove the further sampling. Then, if we find we are reporting too many errors, we can reintroduce the additional sampling.

Reply:
The comment in `raven.js` looks incorrect to me. I think 0.008 is equivalent to 0.8%. I'll change the comment and leave the rate as it currently is.